src/rpc/daemon_handler.cpp

```diff
@@ -0,0 +1,881 @@
+    {
+      res.status = Message::STATUS_FAILED;
+      res.error_details = "Requested block does not exist";
+      return;
+    }
+
+    res.status = Message::STATUS_OK;
+  }
+
+  void DaemonHandler::handle(const GetBlockHeadersByHeight::Request& req, GetBlockHeadersByHeight::Response& res)
+  {
+    res.headers.resize(req.heights.size());
+
+    for (size_t i=0; i < req.heights.size(); i++)
+    {
+      const crypto::hash block_hash = m_core.get_block_id_by_height(req.heights[i]);
+
+      if (!getBlockHeaderByHash(block_hash, res.headers[i]))
+      {
+        res.status = Message::STATUS_FAILED;
+        res.error_details = "A requested block does not exist";
+        return;
+      }
+    }
+
+    res.status = Message::STATUS_OK;
+  }
+
+  void DaemonHandler::handle(const GetBlock::Request& req, GetBlock::Response& res)
+  {
+    res.status = Message::STATUS_FAILED;
+    res.error_details = "RPC method not yet implemented.";
+  }
+
+  //TODO: this RPC call is marked for later implementation in the old RPC,
+  //      need to sort out if it's necessary and what it should do.
+  void DaemonHandler::handle(const GetPeerList::Request& req, GetPeerList::Response& res)
+  {
+    res.status = Message::STATUS_FAILED;
+    res.error_details = "RPC method not yet implemented.";
+  }
+
+  void DaemonHandler::handle(const SetLogHashRate::Request& req, SetLogHashRate::Response& res)
+  {
+    res.status = Message::STATUS_FAILED;
+    res.error_details = "RPC method not yet implemented.";
+  }
+
+  void DaemonHandler::handle(const SetLogLevel::Request& req, SetLogLevel::Response& res)
+  {
+    if (req.level < 0 || req.level > 4)
+    {
+      res.status = Message::STATUS_FAILED;
+      res.error_details = "Error: log level not valid";
+    }
+    else
+    {
+      res.status = Message::STATUS_OK;
+      mlog_set_log_level(req.level);
+    }
+  }
+
+  void DaemonHandler::handle(const GetTransactionPool::Request& req, GetTransactionPool::Response& res)
+  {
+    bool r = m_core.get_pool_for_rpc(res.transactions, res.key_images);
+
+    if (!r) res.status = Message::STATUS_FAILED;
+    else res.status = Message::STATUS_OK;
+  }
+
+  void DaemonHandler::handle(const GetConnections::Request& req, GetConnections::Response& res)
+  {
+    res.status = Message::STATUS_FAILED;
+    res.error_details = "RPC method not yet implemented.";
+  }
+
+  void DaemonHandler::handle(const GetBlockHeadersRange::Request& req, GetBlockHeadersRange::Response& res)
+  {
+    res.status = Message::STATUS_FAILED;
+    res.error_details = "RPC method not yet implemented.";
+  }
+
+  void DaemonHandler::handle(const StopDaemon::Request& req, StopDaemon::Response& res)
+  {
+    res.status = Message::STATUS_FAILED;
+    res.error_details = "RPC method not yet implemented.";
+  }
+
+  void DaemonHandler::handle(const FastExit::Request& req, FastExit::Response& res)
+  {
+    res.status = Message::STATUS_FAILED;
+    res.error_details = "RPC method not yet implemented.";
+  }
```

Comment:
I'll go through all the calls to make sure they're up to date, and get rid of the deprecated ones.